pkg/controller/machine_test.go

```diff
@@ -183,55 +184,59 @@ func TestController_AddDeleteFinalizerOnlyIfValidationSucceeded(t *testing.T) {
 		cloudProviderSpec string
 		finalizerExpected bool
 		err               string
+		expectedActions   []string
```

Comment:
Wouldnt it make more sense to define this as int, if we only use its length anyways?

Reply:
ups, let me just update the code to compare actual action's verbs with the expected ones.

Reply:
I've just added the following code to verify the verbs of actions.

```
for index, action := range fakeMachineClient.Actions() {
	if !action.Matches(test.expectedActions[index], "machines") {
		t.Fatalf("unexpected action %#v", action)
	}
}
```

Reply:
Nice! When are you going to to the "How to test k8s controllers" workshop? :P

Reply:
don't know if I could show anything more :P